dotnet/Data/WishListRepository.cs

```diff
@@ -244,19 +256,73 @@ public async Task<WishListsWrapper> GetAllLists()
                 request.Headers.Add(WishListConstants.VtexIdCookie, authToken);
                 request.Headers.Add(WishListConstants.PROXY_AUTHORIZATION_HEADER_NAME, authToken);
             }
-            
             request.Headers.Add("Cache-Control", "no-cache");
+            var response = await client.SendAsync(request);
+            tokenResponse = response.Headers.GetValues("X-VTEX-MD-TOKEN").FirstOrDefault();
+
+            string responseContent = await response.Content.ReadAsStringAsync();
+            _context.Vtex.Logger.Debug("GetAllLists", null, $"[{response.StatusCode}]");
+
+            return responseContent;
+        }
 
+        private async Task<string> subScroll()
+        {
             var client = _clientFactory.CreateClient();
+            var request = new HttpRequestMessage
+            {
+                Method = HttpMethod.Get,
+                RequestUri = new Uri($"http://{this._httpContextAccessor.HttpContext.Request.Headers[WishListConstants.VTEX_ACCOUNT_HEADER_NAME]}.vtexcommercestable.com.br/api/dataentities/{WishListConstants.DATA_ENTITY}/scroll?_token={tokenResponse}")
+            };
+
+            string authToken = this._httpContextAccessor.HttpContext.Request.Headers[WishListConstants.HEADER_VTEX_CREDENTIAL];
+            if (authToken != null)
+            {
+                request.Headers.Add(WishListConstants.AUTHORIZATION_HEADER_NAME, authToken);
+                request.Headers.Add(WishListConstants.VtexIdCookie, authToken);
+                request.Headers.Add(WishListConstants.PROXY_AUTHORIZATION_HEADER_NAME, authToken);
+            }
+            request.Headers.Add("Cache-Control", "no-cache");
             var response = await client.SendAsync(request);
+
```

Comment:
Does the token need to be reset?

Reply:
According to the scroll document [https://developers.vtex.com/vtex-rest-api/reference/scrolldocuments](url), all subsequent requests use the same  token from the first call